.github/workflows/pr.yaml

```diff
@@ -1,9 +1,9 @@
-name: GitHub CI
+name: PR commit checks and PR target branch commit checks
 
 on:
   push:
-    # test this branch and staged PRs based on this branch code
-    branches: [ "master", "auto" ]
+    # test commits on this branch
+    branches: [ "master" ]
```

Comment:
I think I understand what you mean. My understanding was that
since we test the PR when it is updated, there is no need to re-run
these tests since they've been already run.
But then this would miss running these tests after any other PR
has been rebased before this. Good point, re-adding the "auto"
branch here is the easiest, safest, and least expensive option

Reply:
> My understanding was that since we test the PR when it is updated, there is no need to re-run these tests ... But then this would miss running these tests after any other PR has been rebased before this.

If you replace "rebased" with "merged", then this would be more-or-less accurate. Here is a simplified summary:

* The latest PR branch commit X is a combination of master branch _as it existed at PR branching time_ and all PR branch commits. Our current CI does not test these commits, but we often talk as if it does.
* So called GitHub "merge commit" Y is a combination of master branch _as it existed at the last PR branch update time_ IIRC and all PR branch commits. Jenkins and GitHub Actions test these commits when they test "PR".
* A staged commit Z is a combination of master branch _as it existed when Z was staged_[^1] and all PR branch commits. Jenkins and GitHub Actions test these commits when they test "auto" branch.

[^1]: Z's parent commit is then-current master. If the corresponding PR is merged, master becomes Z.

Thus, the relevant difference here is the difference between Y and Z. That difference is all commits that happened on master since PR branch was last updated. If a given PR stays constant, that difference grows with every merged PR.